### PR TITLE
[testapp] fix example app build for playstore publish

### PR DESCRIPTION
**Pull request by ank27**
### Summary of changes
- Fixes playstore error to target api 31 (see screenshot).
- Update `tagsize` in gradle script to fix automatic deploy to playstore

Screenshot 2022-11-17 at 4 37 13 PM

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: buildSrc/src/main/kotlin/Project.kt [hunk 1]
```diff
@@ -12,6 +12,11 @@ object AndroidVersions {
     const val targetSdkVersion = 30
     const val compileSdkVersion = 31
   }
+  object ExampleApp {
```
Comment (yunikkk) [inline buildSrc/src/main/kotlin/Project.kt:15]: Should we migrate everywhere to target 31 if Google Play now enforces it?

Comment (ank27) [reply buildSrc/src/main/kotlin/Project.kt:15]: for sdk we can keep as it is, since its upto developers who uses maps-sdk to use targetSdk as 31 for app to deploy. but we can definitely update the downstream app's target version to 31. I will update those.

File: buildSrc/src/main/kotlin/Project.kt [hunk 2]
```diff
@@ -12,6 +12,11 @@ object AndroidVersions {
...
+    const val minSdkVersion = 23
+    const val targetSdkVersion = 31
+    const val compileSdkVersion = 31
```
Comment (alexander-kulikovskii) [inline buildSrc/src/main/kotlin/Project.kt:18]: I guess nice to tests with the latest version.
```suggestion
    const val targetSdkVersion = 33
    const val compileSdkVersion = 33
```

File: gradle/script-git-version.gradle [hunk 3]
```diff
@@ -11,7 +11,7 @@ import org.ajoberstar.grgit.Grgit
 
 ext {
     git = Grgit.open(currentDir: projectDir)
-    def tagSize = git.tag.list().findAll { (it.name =~ "android-v*" || it.name =~ "v1\\d.*") }.size()
+    def tagSize = git.tag.list().findAll { (it.name =~ "android-v*" || it.name =~ "v1\\d.*") }.size() + 1
```
Comment (alexander-kulikovskii) [inline gradle/script-git-version.gradle:14]: why did you add `+1`?

**Comment by yunikkk** [comment]
@ank27 since Play Store no longer accepts apps with targetSdk < 31 - should be migrate all of our apps? Would be helpful to users to see them with the correct and fresh target.

File: buildSrc/src/main/kotlin/Project.kt (hunk 1 above)
Comment (yunikkk) [reply buildSrc/src/main/kotlin/Project.kt:15]: Not sure I understand - why do we want to keep outdated target for the SDK? I assume we might need to update some parts of the SDK as well to correctly handle and adopt all of the requirements of the newer target. 
https://developer.android.com/google/play/requirements/target-sdk from the list of requirements it seems that we need to make sure `location permission` and `intent filters` sections are correctly implemented in the SDK, otherwise users might have problems I assume.

Comment (ank27) [reply buildSrc/src/main/kotlin/Project.kt:15]: > I assume we might need to update some parts of the SDK as well to correctly handle and adopt all of the requirements of the newer target.

yeah, I have this concern only and upgrading target of sdk is not in scope of this pr. 
created [ticket](https://mapbox.atlassian.net/browse/MAPSAND-651) to track this

File: gradle/script-git-version.gradle (hunk 3 above)
Comment (ank27) [reply gradle/script-git-version.gradle:14]: we provided an extra build to rave team manually. this was added to remove the error regarding versionCode should be greater than older versionCode in playstore

Comment (alexander-kulikovskii) [reply gradle/script-git-version.gradle:14]: ah. got it
maybe you should add extra comment here

**Review by alexander-kulikovskii — APPROVED** [review]
LGTM with small question
